command/certificate/create.go

```diff
@@ -707,17 +764,25 @@ func parseSigner(ctx *cli.Context, defaultSigner crypto.Signer) (*x509.Certifica
 }
 
 // savePrivateKey saves the given key, asking the password if necessary.
-func savePrivateKey(filename string, priv interface{}, insecure bool) error {
+func savePrivateKey(ctx *cli.Context, filename string, priv interface{}, insecure bool) error {
+	var err error
 	if insecure {
-		_, err := pemutil.Serialize(priv, pemutil.ToFile(filename, 0600))
+		_, err = pemutil.Serialize(priv, pemutil.ToFile(filename, 0600))
 		return err
 	}
 
 	var pass []byte
-	pass, err := ui.PromptPassword("Please enter the password to encrypt the private key",
-		ui.WithValidateNotEmpty())
-	if err != nil {
-		return errors.Wrap(err, "error reading password")
+	if ctx.IsSet("password-file") {
```

Comment:
Is probably better to check for empty string 
```go
if passwordFile := ctx.String("password-file"); passwordFile != "" {
    // ... 
}
```

`--password-file ""` marks password-file as set.